jni/src/common/Foreign/JNI/Unsafe/Introspection.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Foreign.JNI.Unsafe.Introspection
+ ( getClassName
+ , getSignatures
+ , toText
+ , classGetNameMethod
+ ) where
```

Comment:
Unless you want this interface to be public, I'd name it `Foreign.JNI.Unsafe.Internal.Introspection`, and I would only export `getSignatures`.

Reply:
This module is probably too low-level to be made public, let's move it.
However, we need at least `getSignatures` and `getClassName` to build the Exception message. `classGetNameMethod` is used in the tests ; I figured it would be simpler to export it than to rebuild it. `toText` can be removed.